README.md

```diff
@@ -1,36 +1,282 @@
+  ios.assets "resources/ios"
+end
+```
+
+When `hi5 deploy ios` is ran, high five will include `resources/ios` in the destination.
+
+For each platform, you need to make sure `config/high_five/app-<platform>.js` exists.
+See current examples for the minimum contents required in these files.
+This is important for INSERT REASONS HERE.  (TODO Why not just say `ios.javascripts "ios_specific_javascript_file.js"` instead?)
+
+* `config.environment`
+
+Environments function like platforms: they allow you to further specify or override either base or platform-specific settings for each of your environments (production/qa/staging/etc).
+That is, they take precedence over platform configurations.
+
+Their primary purpose is to set your app's endpoint, but can be used for anything.
+e.g. You may want a non-minified asset deployed to your device during development, but the minified version deployed all other times.
+
+An environment is specified with either `-e` or `--environment` when running `hi5 deploy`.
+
+e.g.
+
+  `hi5 deploy <platform> --environment lan`
+
+  `hi5 deploy <platform> -e production`
+
+At a minimum, you will probably want `:lan` (so you can deploy an app which is pointed at a server running on your machine) and an entry for each environment your app has (production/staging/qa/etc).
+```ruby
+config.environment :lan do |lan|
+  ip = Socket.ip_address_list.detect { |intf| intf.ipv4? && !intf.ipv4_loopback? && !intf.ipv4_multicast? }.ip_address
+  lan.setting apiEndpoint: "http://#{ip}:3000" # Or however you specify your server's location.
+  # lan.javascripts "resources/vendor/javascripts/some-library-debug-all.js"
+end
+
+config.environment :dev do |dev|
+  dev.setting apiEndpoint: "http://dev.example.com"
+  # dev.javascripts "lib/some-library-minified.js"
+end
+
+config.environment :production do |production|
+  production.setting apiEndpoint: "http://example.com"
+  # production.javascripts "lib/some-library-minified.js"
+  # production.minify :uglifier # or :yui
+end
+```
+
+## Usage
+
+* `$ hi5 deploy <platform> -e <environment>`
+
+This will build your application for a specific platform with the settings for the given environment to the destination directory specified in `config/high_five.rb`.
+
+* There's a set icon task `$ hi5 <platform>:set_icon path/to/icon.png` I think?.
+
+TODO: Is it really <platform> or just android/ios?
+
+By default, high_five uses ChunkyPNG to resize the icon, but this almost always creates shitty looking icons.
+We highly recommend adding the [rmagick](https://github.com/rmagick/rmagick) gem (which depends on imagemagick) to your Gemfile along with high five.
+High Five will detect that you have rmagick installed and will use it to generate much nicer icons.
+
+TODO: Just make rmagick a dependency since ChunkyPNG is shit and no one will want to use it anyways?
+
+* Pretty sure there's a task to set version? But I don't really know what that means, what it does, how it works, or how to run it.
```

Comment:
`high_five (ios|android):set_version -v "1.0" -b 200`
This sets the publicly visible "version string" like 1.0 or whatever and also the internal build number.  They are held either in the info.plist for ios or androidmanifest.xml.  They are both important